src/NServiceBus.TransportTests/NServiceBusTransportTest.cs

```diff
@@ -20,6 +20,9 @@ public abstract class NServiceBusTransportTest
         public void SetUp()
         {
             testId = Guid.NewGuid().ToString();
+
+            //when using [TestCase] NUnit will reuse the same test instance so we need to make sure that the message pump is a fresh one
+            MessagePump = null;
```

Comment:
this is quite a hack, but we decided to go with that in this PR as it's not this PR's fault and instead raise an issue to find a proper solution to this problem: https://github.com/Particular/NServiceBus/issues/4675